src/types.ts

```diff
@@ -279,11 +279,11 @@ export interface SharedProps<V extends VariantType = VariantType> extends Partia
      * @param {object} event The event source of the callback
      * @param {string} reason Can be:`"timeout"` (`autoHideDuration` expired) or: `"maxsnack"`
      * (snackbar was closed because `maxSnack` has reached) or: `"instructed"` (snackbar was
-     * closed programmatically)
+     * closed programmatically) or: any string if closeSnackbar had reason parameter
      * @param {string|number|undefined} key key of a Snackbar. key will be `undefined` if closeSnackbar
      * is called with no key (user requested all the snackbars to be closed)
      */
-    onClose?: (event: React.SyntheticEvent<any> | null, reason: CloseReason, key?: SnackbarKey) => void;
+    onClose?: (event: React.SyntheticEvent<any> | null, reason: CloseReason | string, key?: SnackbarKey) => void;
```

Comment:
The reason I didn't edit the `CloseReason` is to think of it as "Official" reasons - something that's defined by the library itself and used internally.
Also, my thinking was that TS would auto-suggest the `CloseReasons` with intellisense - but also allow for any other string, but TS compiler seems to simplify it to string 🤔  https://github.com/Microsoft/TypeScript/issues/29729

Workaround would be to change the type to `reason: CloseReason | (string & { _intellisense: never })`, but felt too hacky to include this in a PR, but it does work:
![image](https://github.com/user-attachments/assets/c5a997fb-e855-495c-a827-af99560333c0)